ICAFIX/scripts/functionhighpassandvariancenormalize.m

```diff
@@ -175,8 +208,8 @@ function functionhighpassandvariancenormalize(TR,hp,fmri,WBC,varargin)
     ciftisave(BO,[fmri '_Atlas' regstring hpstring '.dtseries.nii'],WBC);
     delete('Atlas.nii.gz');
 
-elseif hp<0  % If no hp filtering, still need to at least demean the volumetric time series
-    if dovol > 0
+elseif hp < 0  % If no hp filtering, still need to at least demean the volumetric time series, assumes concatenated input
```

Comment:
One could choose to run single-run with no hp filtering, so the comment added here of "assumes concatenated input" isn't appropriate.
```suggestion
elseif hp < 0  % If no hp filtering, still need to at least demean the volumetric time series
```

Reply:
If you set the current code to do no hp filtering, it currently assumes it is a concatenated run, so basically, no you can't tell it to do that (you might want to, but the code will act differently than you want).  If this specific conditional doesn't contain such assumptions, it could be removed from this comment.

Reply:
hp<0 does not _have_ to mean concatenated data. The previous and existing code code supports use of hp<0 on single run data as well.  Given that, I find the introduction of the `singlerun` variable as a proxy for "hp > 0" to not be particularly clarifying.  How about the following instead:
`dohp = pdflag || (hp >= 0);`
and then remove the comments you added regarding single vs. concatenated runs?

Reply:
This comment block already existed:

https://github.com/Washington-University/HCPpipelines/blob/ba9369d068c7ea510b79f42cb5247e8e8549d5be/ICAFIX/scripts/functionhighpassandvariancenormalize.m#L130

In particular, if hp < 0, it completely skips loading the cifti file, so none of the code that works on cifti can be run.  The only obvious way to make it clear is to change the arguments so that the user can separately specify single run or concatenated, and whatever hp value.  At that point, because virtually no lines in this file are executed by both hp < 0 and hp >= 0 (aka concatenated and single run) cases, it might as well be split into two different functions.  We could take the time and testing to do that, and it would probably improve the code.

Reply:
I have clarified this comment to indicate that the cifti and motion confounds will not have been loaded when this branch is taken.

Reply:
FWIW, I still think the comment revisions/additions on this issue conflate two different things: Saying in the pre-PR version that `functionhighpassandvariancenormalize` expects that `hcp_multi_run_fix` will pass in concatenated data with `hp < 0` is not the same as assuming that "hp < 0 is equivalent to concatenated input".

Reply:
functionhighpass... doesn't currently have a separate flag to convey concatenated/single input mode, and it does in fact currently assume that if hp < 0, it is being run on concatenated data.  It is not assuming they are conceptually equivalent, it is saying that the current logic of the code assumes that each implies the other.  I did not change the logic, that would be more invasive and need more testing, I just substituted a different variable so that we didn't have "hp < 0" on conditionals that actually only care whether it is running in concatenated mode.

Reply:
This version is not general, which is the whole point of these comments.  The comments are specifically pointing out that it can't currently be used to do single run without hp, because the code *makes assumptions* that it probably shouldn't.  That is why the comments I added use the word "assumes".  They are documenting what assumptions it *does* make, right now.  I am not defending them, I am pointing them out so they are less likely to stab someone in the face.

If you think there is a clearer way to convey this shortcoming, then suggest the phrasing you would use.  The comments should match what the code actually is, not what it could be in the future, so your criticism seems off track.

Reply:
I'm not convinced that the function couldn't in fact be used "as is" to run a single run with no HP.  Why do you say that it can't?  Personally, I would use a variable called `doHP` rather than `singlerun` (which is just a convenience variable to avoid having to check for `pdflag || (hp >= 0)`, and remove the comments about assumed concatenated input.

Reply:
If you do give hp = -1, it will not currently open the cifti file (or motion confounds, which are a different story), and if you don't open that, you can't write out a cifti VN map, which is, as I understand it, part of what single-run mode's requirements are.  Thus, in order to actually tell it to do something different in single-run no-hp mode, compared to concat-run no-hp mode, an additional mode must be conveyed to the code somehow, and the code changed to make use of it.

Reply:
The entire reason I made the "singlerun" variable is because some conditionals contain code that is *not* related to the hp setting, but in fact is relevant to single-run versus concat-run modes.  Those conditionals get the "singlerun" test so that they make more sense.  Saying "if doHP, then do this single-run relevant code, otherwise do the concat-run code" would be a regression to the confusion that I specifically made the "singlerun" variable to clarify.

Reply:
Ok.  The lack of a cifti VN map with hp < 0 may be relevant.  Maybe add a comment along those lines to the code, as a reminder if/when we come back to this in the future.

Reply:
Maybe with a link as well referring back to your previous comment.

Reply:
A later conditional specifically doesn't even write out a volume VN map in concat-run mode.  The only unifying principle in these conditionals is just that, single-run versus concat-run mode, so your suggestion amounts to adding a unique comment to every conditional saying what the code inside it does, which seems excessive.  I consider the new "singlerun" variable to make those conditionals closer to self-documenting, unlike their previously misleading test of the highpass setting.

Reply:
To be clear, as far as I can tell, the only conditional in the entire function that should actually care about the high pass setting is the first one that actually does the high pass filtering, and the rest are solely introducing quirks that differentiate the concat-run and single-run mode computation/outputs.  None of them should have ever depended on the `hp` setting in the first place, instead they should have used a new input argument to change the mode.

Reply:
I've suggested one additional line to go with your "HACK" comment, which I think is all that is necessary regarding my request. (Don't think a unique comment is necessary for every conditional).